zmq-lwt/src/deferred.ml

```diff
@@ -2,7 +2,7 @@ type 'a t = 'a Lwt.t
 module Deferred = struct
   type 'a t = 'a Lwt.t
   let return a = Lwt.return a
-  let catch f = Lwt_result.catch (f ())
+  let catch f = Lwt.catch (fun () -> Lwt_result.ok (f ())) Lwt_result.fail
```

Comment:
Ah, that's pretty nice. Instead of requiring the new `Lwt_result.catch` which breaks the API you implement it using `Lwt.catch` and stay compatible with the older versions.

I have an suggestion: define a function `lwt_result_catch` (or `Lwt_result'.catch` even) that implements the "new" `Lwt_result.catch` interface, so in case we ever bump the maximum version of Lwt to 6.0 we can just remove that function and replace it with a call to `Lwt_result.catch`.

This has worked pretty well in other projects I maintain, since when bumping dependency versions I could easily identify code that can be removed.

Reply:
Thanks for the suggestion. I've updated the PR.